lib/ddtrace/profiling/scheduler.rb

```diff
@@ -0,0 +1,66 @@
+require 'ddtrace/worker'
+require 'ddtrace/workers/polling'
+
+module Datadog
+  module Profiling
+    # Pulls and exports profiling data on an async interval basis
+    class Scheduler < Worker
+      include Workers::Polling
+
+      attr_reader \
+        :exporters,
+        :recorder
+
+      def initialize(recorder, exporters, options = {})
+        @recorder = recorder
+        @exporters = [exporters].flatten
+
+        # Workers::Async::Thread settings
+        # Restart in forks by default
+        self.fork_policy = options.fetch(:fork_policy, Workers::Async::Thread::FORK_POLICY_RESTART)
+
+        # Workers::IntervalLoop settings
+        self.loop_base_interval = options[:interval] if options.key?(:interval)
+        self.loop_back_off_ratio = options[:back_off_ratio] if options.key?(:back_off_ratio)
+        self.loop_back_off_max = options[:back_off_max] if options.key?(:back_off_max)
```

Comment:
IIUC, that's a backoff strategy for when there's nothing to do.

You actually want the profiler to really upload every minute and not wait any longer than 60s even if the queue is empty.

In practice, the queue should never be empty, so that's unlikely a problem, but we might want to be able to "disable" this entirely here.

Reply:
Yeah, I think we can tweak the knobs to prevent it from backing off. Can you describe exactly how the timing aspect should work?

Reply:
As its written right now, the looping behavior checks for `work_pending?`, and if `true`, it will set the base interval. Because `Scheduler` does not define `work_pending?` it defaults to `run_loop?` which will always be `true` unless the `Scheduler` is instructed to stop. This means it effectively polls on a single interval without backing off.

Reply:
So that's fine, I'd just remove those lines if possible to make sure it's clear there's no backing off involved. :)

Reply:
One thing that you may want to have and that I recently improved:
https://github.com/DataDog/dd-trace-py/pull/1386

As we want to have a close as possible slice of one minute of event, if your export process takes 10 seconds to complete, the next upload will get 1 minutes and 10 seconds of data.
Better wake up 10 seconds earlier then.

The idea is to meter the time took by the `perform` method and substract it from the sleep interval.

Reply:
Sure, in this case we can probably do this by measuring start-to-end then `loop_back_off!` at the end to set the interval. I'll figure out these details to make this work.

Reply:
Should be addressed.